clang/test/CodeGenSYCL/inline_asm.cpp

```diff
@@ -1,4 +1,4 @@
-// RUN: %clang_cc1 -fsycl-is-device -triple spir64-unknown-unknown -opaque-pointers -emit-llvm -x c++ %s -o - | FileCheck %s
+// RUN: %clang_cc1 -O1 -fsycl-is-device -triple spir64-unknown-unknown -opaque-pointers -emit-llvm -x c++ %s -o - | FileCheck %s
```

Comment:
If default level was O2 prior this patch, why -O1 or -O3 is added to the tests. I'm also wondering why so many changes in emitted IR?

Reply:
> If default level was O2 prior this patch, why -O1 or -O3 is added to the tests? 

Here O2 is not necessary and O1 is good enough to produce the same IR as it was with O2, but test works faster.

> I'm also wondering why so many changes in emitted IR?

The reason is > 100 LLVM IR transformation passes applied with O2.